ocaml/runtime/roots_nat.c

```diff
@@ -80,21 +80,50 @@ static link* frametables_list_tail(link *list) {
   return tail;
 }
 
+/* Special marker instead of frame_size for frame_descr in long format */
+static uint32_t LONG_FRAME = 0x7FFF;
+
+uint32_t get_frame_size(frame_descr *d) {
+  /* CR gyorsh: where is the special frame size 0xFFFF emitted that marks the top of
+   an ML stack chunk? */
```

Comment:
The special frame size 0xFFFF is never emitted, it's hardcoded in `caml_system__frametable` in `runtime/amd64.S`. This defines the frame tables for `caml_start_program`, which is always at the top of an ML stack chunk.